### PR TITLE
[keyboard] Clueboard 2x1800 2021 Support

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

Support for the 2021 Clueboard 2x1800 "LED Sign Edition"

## Types of Changes

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
You got changes to the submodules.

**Review by drashna — APPROVED** [review]

**Comment by skullydazed** [comment]
> You got changes to the submodules.

Fixed, and also rebased onto master.

File: keyboards/clueboard/2x1800/2021/max7219.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#    define MAX7219_INTENSITY 1
+#endif
+
+#ifndef MAX7219_SCROLL_TIME
+#    define MAX7219_SCROLL_TIME 100
+#endif
+
+#ifndef MAX7219_BUFFER_MULTIPILER
+#    define MAX7219_BUFFER_MULTIPILER 24
+#endif
```
Comment (gcochard) [inline keyboards/clueboard/2x1800/2021/max7219.h:51]:
```suggestion
#ifndef MAX7219_BUFFER_MULTIPLIER
#    define MAX7219_BUFFER_MULTIPLIER 24
#endif
```

File: keyboards/clueboard/2x1800/2021/max7219.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#ifndef MAX7219_LOAD
+#    define MAX7219_LOAD B0
+#endif
+
+#ifndef MAX7219_CONTROLLERS
+#    define MAX7219_CONTROLLERS 4
+#endif
+
+#ifndef MAX7219_LED_INTENSITY
+#    define MAX7219_INTENSITY 1
```
Comment (gcochard) [inline keyboards/clueboard/2x1800/2021/max7219.h:42]:
```suggestion
#    define MAX7219_LED_INTENSITY 1
```

**Review by gcochard — CHANGES_REQUESTED** [review]
Had to make these changes to compile the firmware.

File: keyboards/clueboard/2x1800/2021/max7219.h (hunk 2 above)
Comment (skullydazed) [reply keyboards/clueboard/2x1800/2021/max7219.h]: Fixed this and the buffer multiplier. Thanks!

File: keyboards/clueboard/2x1800/2021/keymaps/signboard/keymap.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                max7219_message_sign(kmi_msg, MSG_KONAMI_LEN);
+                return true;
+            case MSG_QMK:
+                max7219_led_scrolling=true;
+                uint8_t qmk_msg[MSG_QMK_POWERED_LEN][6] = MSG_QMK_POWERED;
+                max7219_message_sign(qmk_msg, MSG_QMK_POWERED_LEN);
+                return true;
+        }
+    }
+    return false;
```
Comment (gcochard) [inline keyboards/clueboard/2x1800/2021/keymaps/signboard/keymap.c:49]: If this is false, no key presses work. According to the [macro docs][1], returning `true` is the correct way to signal to the fw that the keypress should be processed normally.
```suggestion
    return true;
```

[1]: https://docs.qmk.fm/#/feature_macros

File: keyboards/clueboard/2x1800/2021/keymaps/default_4u/keymap.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_4u_space(
+      KC_HOME, KC_END,  KC_PGUP, KC_PGDN,       KC_ESC,    KC_F1,   KC_F2,   KC_F3,   KC_F4,     KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,   KC_F10,   KC_F11,  KC_F12,        KC_PSCR, KC_SLCK, KC_PAUS, KC_INS,  \
+                                                                                                                                                                                                                      \
+      KC_PMNS, KC_NLCK, KC_PSLS, KC_PAST,      KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC,     KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, \
+      KC_PPLS, KC_P7,   KC_P8,   KC_P9,        KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,       KC_P7, KC_P8, KC_P9, KC_PSLS,       \
```
Comment (gcochard) [inline keyboards/clueboard/2x1800/2021/keymaps/default_4u/keymap.c:23]:
```suggestion
      KC_PPLS, KC_P7,   KC_P8,   KC_P9,        KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,       KC_P7, KC_P8, KC_P9, KC_PPLS,       \
```

File: keyboards/clueboard/2x1800/2021/keymaps/default_7u/keymap.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_7u_space(
+      KC_HOME, KC_END,  KC_PGUP, KC_PGDN,       KC_ESC,    KC_F1,   KC_F2,   KC_F3,   KC_F4,     KC_F5,   KC_F6,   KC_F7,   KC_F8,     KC_F9,   KC_F10,   KC_F11,  KC_F12,        KC_PSCR, KC_SLCK, KC_PAUS, KC_DEL,  \
+                                                                                                                                                                                                                      \
+      KC_PMNS, KC_NLCK, KC_PSLS, KC_PAST,      KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC,     KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, \
+      KC_PPLS, KC_P7,   KC_P8,   KC_P9,        KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,       KC_P7, KC_P8, KC_P9, KC_PSLS,       \
```
Comment (gcochard) [inline keyboards/clueboard/2x1800/2021/keymaps/default_7u/keymap.c:23]:
```suggestion
      KC_PPLS, KC_P7,   KC_P8,   KC_P9,        KC_TAB,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_LBRC, KC_RBRC, KC_BSLS,       KC_P7, KC_P8, KC_P9, KC_PPLS,       \
```

**Review by gcochard — CHANGES_REQUESTED** [review]

File: keyboards/clueboard/2x1800/2021/keymaps/signboard/keymap.c (hunk 3 above)
Comment (skullydazed) [reply keyboards/clueboard/2x1800/2021/keymaps/signboard/keymap.c]: Thanks!

**Pull request merged**